app/models/presentation.rb

```diff
@@ -1,4 +1,24 @@
 class Presentation < ApplicationRecord
-  has_many :person_presentations
-  has_many :people, through: :person_presentations
+  belongs_to :person, optional: true
+  belongs_to :event
+
+  validates :topic,     presence: true
+  validates :duration,  presence: true
+  validates :person,    presence: true, unless: :presenter
+  validates :presenter, presence: true, unless: :person
+  validates :event,     presence: true
+
+  validate  :person_or_presenter
+
+  def presenter_name
+    person ? person.first_name : presenter
+  end
+
+  private
+
+  def person_or_presenter
+    return unless person.present? && presenter.present?
```

Comment:
I always found `unless` to read a little funny. Worth mentioning that you could also do
```
return if person.nil? && presenter.nil?
```